mav_trajectory_generation/src/polynomial.cpp

```diff
@@ -20,8 +20,113 @@
 
 #include "mav_trajectory_generation/polynomial.h"
 
+#include <algorithm>
+#include <limits>
+
 namespace mav_trajectory_generation {
 
+void Polynomial::findMinMaxCandidates(
+    double t_start, double t_end,
+    const Eigen::VectorXcd& roots_derivative_of_derivative,
+    std::vector<double>* candidates) const {
+  CHECK_NOTNULL(candidates);
+  candidates->clear();
+  candidates->reserve(roots_derivative_of_derivative.size() + 2);
+  candidates->push_back(t_start);
+  candidates->push_back(t_end);
+  for (size_t i = 0; i < roots_derivative_of_derivative.size(); i++) {
+    // Only real roots are considered as critical points.
+    if (std::abs(roots_derivative_of_derivative[i].imag()) >
+        std::numeric_limits<double>::epsilon()) {
+      continue;
+    }
+    const double candidate = roots_derivative_of_derivative[i].real();
+    // Do not evaluate points outside the domain.
+    if (candidate < std::min(t_start, t_end) ||
+        candidate > std::max(t_start, t_end)) {
+      continue;
+    } else {
+      candidates->push_back(candidate);
+    }
+  }
+}
+
+bool Polynomial::findMinMaxCandidates(double t_start, double t_end,
+                                      int derivative,
+                                      std::vector<double>* candidates) const {
+  CHECK_NOTNULL(candidates);
+  candidates->clear();
+  if (N_ - derivative - 1 < 0) {
+    LOG(WARNING) << "N - derivative - 1 has to be at least 0.";
+    return false;
+  }
+  Eigen::VectorXcd roots_derivative_of_derivative;
+  if (!findRootsJenkinsTraub(getCoefficients(derivative + 1),
+                             &roots_derivative_of_derivative)) {
+    return false;
+  } else {
+    findMinMaxCandidates(t_start, t_end, roots_derivative_of_derivative,
+                         candidates);
+    return true;
+  }
+}
+
+bool Polynomial::findMinMax(
+    double t_start, double t_end, int derivative,
+    const Eigen::VectorXcd& roots_derivative_of_derivative,
+    std::pair<double, double>* min, std::pair<double, double>* max) const {
```

Comment:
Probably best to avoid using min and max as variable names, might shadow std::min/max if `using namespace std;`. Fine if you want to keep it though.